### PR TITLE
TGC-476 + TGC-479 page templating yes no question

### PR DESCRIPTION
**Pull request by Andrew-Folga**
This PR introduces following changes:

- introduces single nunjucks template for all standard pages called page.njk
- provides implementation for first question (country) with relevant answers and designs
- tracks user answers
- add sidebar macros and implements it on the generic page template
- Improves narrow tests with interaction tests
- changes library from cheerio back to domjs as cheerio does not support interactions

Note: This PR does not repopulate previous answers when traversing pages back and forth yet.

Screenshot 2024-11-07 at 11 31 21

### CODE REVIEW

File: package.json
Comment (ScenarioFactory) [inline package.json:1]: Should there not be a package bump?

File: src/config/machines/example-grant-machine.js [hunk 1]
```diff
@@ -23,6 +22,20 @@ export const actionImplementations = {
     currentPageId: (_context, event) => {
       return event.type === 'NEXT' ? event.nextPageId : event.previousPageId;
     }
+  }),
+  updateAnswers: assign({
+    /**
+     * Updates the completed pages
```
Comment (ScenarioFactory) [inline src/config/machines/example-grant-machine.js:68]: Is that what the function does? Maybe I don't understand these

Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js:68]: yes it just takes the args and defines the new object. It is not mutating the old it is returning new.

File: package.json
Comment (Andrew-Folga) [reply package.json:1]: Done ta

File: src/config/machines/example-grant-machine.js [hunk 2]
```diff
@@ -108,12 +163,25 @@ export const exampleGrantMachine = createMachine({
...
       meta: {
+        id: 'second-question', // NOSONAR:S1192 - need to improve this later
         previousPageId: 'country',
-        nextPageId: 'final'
+        nextPageId: 'final',
+        title: 'Is this a second question?',
+        questionType: 'radio',
+        answers: [
+          {
+            key: 'country-A1',
```
Comment (alanplatt) [inline src/config/machines/example-grant-machine.js]: Should this be second-questions-A1?

File: src/config/machines/example-grant-machine.js [hunk 3]
```diff
@@ -108,12 +163,25 @@ export const exampleGrantMachine = createMachine({
...
+            value: 'Yes'
+          },
+          {
+            key: 'country-A2',
```
Comment (alanplatt) [inline src/config/machines/example-grant-machine.js]: Should this be second-questions-A2?

File: src/http/routes/grant-type/get-context.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { app } from '../../../config/app.js';
  */
 export function getContext(grantTypeId, meta) {
   return {
-    siteTitle: `${app.siteTitle} - ${meta.currentPageId}`,
+    pageTitle: `${app.siteTitle} - ${meta.id}`,
```
Comment (alanplatt) [inline src/http/routes/grant-type/get-context.js]: Was it not clearer with currentPageId?

File: src/views/pages/example-grant/final.njk
Comment (alanplatt) [inline src/views/pages/example-grant/final.njk:1]: Why not make this common to all grants?

File: src/config/machines/example-grant-machine.js (hunk 2 above)
Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js]: Done

File: src/http/routes/grant-type/get-context.js (hunk 4 above)
Comment (Andrew-Folga) [reply src/http/routes/grant-type/get-context.js]: Done

File: src/views/pages/example-grant/final.njk
Comment (Andrew-Folga) [reply src/views/pages/example-grant/final.njk:1]: Done

**Review by alanplatt — DISMISSED** [review]

**Review by ScenarioFactory — APPROVED** [review]

**Pull request merged**